drivers/gpio/gpio_esp32.c

```diff
@@ -83,37 +78,11 @@ static inline u32_t *gpio_pin_reg(int pin)
 	return (u32_t *)(GPIO_PIN0_REG + pin * 4);
 }
 
-static int config_interrupt(u32_t pin, int flags)
-{
-	volatile u32_t *reg = gpio_pin_reg(pin);
-	int type = convert_int_type(flags);
-	u32_t v;
-	unsigned int key;
-
-	if (type < 0) {
-		return type;
-	}
-
-	key = irq_lock();
-
-	v = *reg;
-	v &= ~(GPIO_PIN_INT_ENA_M | GPIO_PIN_INT_TYPE_M);
-	/* Bit 3 of INT_ENA will enable interrupts on CPU 0 */
-	v |= (1<<2) << GPIO_PIN_INT_ENA_S;
-	/* Interrupt triggering mode */
-	v |= type << GPIO_PIN_INT_TYPE_S;
-	*reg = v;
-
-	irq_unlock(key);
-
-	return 0;
-}
-
 static void config_polarity(u32_t pin, int flags)
 {
 	volatile u32_t *reg = (u32_t *)(GPIO_FUNC0_IN_SEL_CFG_REG + pin * 4U);
 
-	if (flags & GPIO_POL_INV) {
+	if (flags & GPIO_ACTIVE_LOW) {
```

Comment:
I wonder if this might be the problem with the test failures.  The wrapper functions in `gpio.h` now take responsibility for handling polarity by inverting various flags and values so that the driver sees everything as though it were active high.  If you're enabling hardware polarity inversion in some circumstances the results will be wrong.

Reply:
as we discussed on slack, i removed `config_polarity` function.